src/components/utils/Markdown.tsx

```diff
@@ -0,0 +1,165 @@
+import { useState, useEffect, useMemo } from "react";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { coy } from "react-syntax-highlighter/dist/esm/styles/prism";
+import remarkGfm from "remark-gfm";
+import styled from "@emotion/styled";
+import { Button, Box } from "@chakra-ui/react";
+import { CopyIcon } from "@chakra-ui/icons";
+import { Row } from "../../utils/chakra";
+import { copySnippetToClipboard } from "../../utils/clipboard";
+
+// Required to display accurate font sizes for markdown elements.
+const StyledMarkdownWrapper = styled(Box)`
+  h1,
+  h2,
+  h3,
+  h4,
+  h5,
+  h6 {
+    font-size: inherit;
+    font-weight: 500;
+  }
+
+  blockquote {
+    margin: revert;
+  }
+
+  ol,
+  ul {
+    margin-left: 20px;
+  }
+  h1 {
+    font-size: 2em;
+  }
+
+  h2 {
+    font-size: 1.5em;
+  }
+
+  h3 {
+    font-size: 1.17em;
+  }
+
+  h4 {
+    font-size: 1em;
+  }
+
+  h5 {
+    font-size: 0.83em;
+  }
+
+  h6 {
+    font-size: 0.67em;
+  }
+
+  hr {
+    background-color: #000000;
+    height: 2px;
+    border: 0px;
+  }
+`;
+
+export interface MarkdownProps {
+  text: string;
+}
+
+const TitleBar = ({ language, code }: { language?: string; code: string }) => {
+  return (
+    <Row
+      mainAxisAlignment="flex-start"
+      crossAxisAlignment="stretch"
+      expand
+      justifyContent="space-between"
+      alignItems="center"
+      px="10px"
+      py="5px"
+      backgroundColor="#f5f5f5"
+      borderBottom="1px solid #eee"
+      borderRadius="6px"
+    >
+      <Box>{language || "plaintext"}</Box>
```

Comment:
we use `plaintext` up here but `text` later in the file?

Reply:
changed `text` to `plaintext` for consistency.